LIBSTELL/Sources/Modules/mgrid_mod.f

```diff
@@ -667,30 +670,65 @@ SUBROUTINE read_mgrid_nc (filename, extcur, nv, nfp,
      1           MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL, 
      2           shar_comm, istat)
          CALL MPI_COMM_RANK(shar_comm, shar_rank, istat)
-      ELSE
-         mpi_rank = 0; mpi_size = 1; 
-         shar_comm = -1; shar_rank = 0
+
       END IF
 #endif
 C-----------------------------------------------
-      call cdf_open(ngrid, filename,'r', istat)
+!
+!     OPEN FILE
+!
+      IF (shar_rank .eq. 0) CALL cdf_open(ngrid, filename,'r', istat)
+#if defined(MPI_OPT)
+      CALL MPI_BCAST(istat,1,MPI_INTEGER,0,shar_comm,MPI_ERR)
+#endif
       IF (istat .ne. 0) THEN
          ier_flag = 9
          RETURN
       END IF
 
 !
-!     READ IN DATA
+!     READ IN BASIC DATA
```

Comment:
@lazersos Have you modified the binary mgrid file part correspondingly? I was using binary mgrid files in the case I showed.

Reply:
Nope only the netCDF part so far.

Reply:
@lazersos With the updates, VMEC2000 is now not able to read MGRID file in binary. It will cause errors with `PMPI_Bcast`.

Reply:
@zhucaoxiang c6d0039 I added the ability to read binary MGRID files as well.  It passed the test on my laptop.  Also it should be faster.

Reply:
@lazersos Just fix this issue. There is an additional line that causes errors.

Reply:
@zhucaoxiang what issue?  I checked and binary files written by MAKEGRID can be read and VMEC executes.

Reply:
@lazersos There is one line I think you forgot to delete causing errors. Check https://github.com/PrincetonUniversity/STELLOPT/pull/79/commits/8f73e7a694b441bdcecbb9f5eaaac8c2d90cc342.